### PR TITLE
IGNITE-21656 :  Cache dump fails on a cache with a node filter. V3

### PR DESCRIPTION
**Pull request by Vladsz83**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java [hunk 1]
```diff
@@ -362,6 +362,9 @@ private Map<PartitionKeyV2, PartitionHashRecordV2> checkDumpFiles(
         SnapshotHandlerContext opCtx,
         Set<File> partFiles
     ) {
+        if (F.isEmpty(partFiles))
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java:365]: It looks like this condition should be outside of the method. If param is empty then no need to check them?

Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java]: Fixed but works wierd. Snapshot contains meta. It is a partition but might be ignored too

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/Dump.java [hunk 2]
```diff
@@ -163,10 +163,12 @@ public Dump(
      * @return Standalone kernal context.
      */
     private GridKernalContext standaloneKernalContext(File dumpDir, IgniteLogger log) {
-        File binaryMeta = CacheObjectBinaryProcessorImpl.binaryWorkDir(dumpDir.getAbsolutePath(), F.first(metadata).folderName());
+        SnapshotMetadata meta = metadata.stream().filter(m -> !m.partitions().isEmpty()).findFirst().orElse(F.first(metadata));
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/Dump.java:166]: Looks overcomplicated.

Comment (timoninmaxim) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/Dump.java:166]: Use F.view() instead

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/Dump.java [hunk 3]
```diff
@@ -163,10 +163,12 @@ public Dump(
...
+
+        File binaryMeta = CacheObjectBinaryProcessorImpl.binaryWorkDir(dumpDir.getAbsolutePath(), meta.folderName());
         File marshaller = new File(dumpDir, DFLT_MARSHALLER_PATH);
 
-        A.ensure(binaryMeta.exists(), "binary metadata directory not exists");
+        A.ensure(meta.partitions().isEmpty() || binaryMeta.exists(), "binary metadata directory not exists");
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/Dump.java:171]: If meta.partitions() is empty, it means dump doesn't have partitions at all - it looks like a bug.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java [hunk 4]
```diff
@@ -362,6 +362,13 @@ private Map<PartitionKeyV2, PartitionHashRecordV2> checkDumpFiles(
         SnapshotHandlerContext opCtx,
         Set<File> partFiles
     ) {
+        if (F.isEmpty(partFiles)) {
+            log.info("Dump data integrity check skipped [snpName=" + opCtx.metadata().snapshotName()
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java:366]: snpName -> dumpName, let's also add param for "this node" - consistentId?.

Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java:366]: Thought about it. All the other code uses 'snp' and 'snapshot' for dumps. But ok

**Pull request closed without merging**